docs/targeting.md

```diff
@@ -0,0 +1,102 @@
+
+#### Cache deletion
+
+Using a dedicated cache object for each disruption works great; the reconcile loop is ran on every possible target events, and the disruption adapts wonderfully to changing targets. Ressource consumption is stable and seems to scale. However when the disruption is deleted, we need to make sure the cache is deleted as well. This proved more difficult than it looked.
+
+When created, the cache has a context, and a cancel function. Those are stored with the cache reference in the aforementionned map, and called upon the final disruption reconcile. However, if the deletion does not go through for any given reason, this cache can be very hard to recover without restarting the controller. To uphold our safety and performance standards, we need safety mechanisms to assert a cache cannot be lost in nature after a disruption is deleted.
+
+There are three folds:
+
+- [Clear Instance Selector Cache](https://github.com/DataDog/chaos-controller/blob/adb8070c989a6c25195354e3bcef3f2c839ef032/controllers/cache_handler.go#L552): This function is called as late as possible before the disruption deletion. It calls the cache cancel function and deletes the cache map entry. This will cover most cases, and following listings are extra safeties -- because the reconcile loop can be interrupted.
+- [Expired Cache Context Cleanup](https://github.com/DataDog/chaos-controller/blob/adb8070c989a6c25195354e3bcef3f2c839ef032/controllers/cache_handler.go#L571): on every reconcile, asserts for every existing cache the corresponding disruption exists. Also, asserts no cache listed in the map has with a already-canceled context - in which case it's deleted.
+- [Cache Deletion Safety](https://github.com/DataDog/chaos-controller/blob/adb8070c989a6c25195354e3bcef3f2c839ef032/controllers/cache_handler.go#L596): the first two methods are useful and will prevent most cases from going wrong, but still rely on the controller looping after the disruption is over to make sure the resources are released. Started in its own goroutine this function will, every minute, poll the disruption. If it is not found the cache context will be canceled, releasing the resources it used. If the controller reconciles again, the second listed function will erase the cache from the map, as the context will be found canceled.
+
+To ensure no cache is running loose, a gauge indicating the amount of current cache entries is set to be monitored; it can be substracted to the gauge of live disruptions to alert in case of sustained negative results.
```

Comment:
```suggestion
To ensure no cache is running loose, a gauge indicating the amount of current cache entries is set to be monitored; it can be subtracted from the gauge of live disruptions to alert in case of sustained negative results.
```